mdit_py_plugins/admon/index.py

```diff
@@ -0,0 +1,190 @@
+# Process admonitions and pass to cb.
+
+from typing import Dict, Optional, Tuple
+import math
+from markdown_it import MarkdownIt
+from markdown_it.rules_block import StateBlock
+
+ADMONITION_TAGS = [
+    "note",
+    "summary",
+    "abstract",
+    "tldr",
+    "info",
+    "todo",
+    "tip",
+    "hint",
+    "success",
+    "check",
+    "done",
+    "question",
+    "help",
+    "faq",
+    "warning",
+    "attention",
+    "caution",
+    "failure",
+    "fail",
+    "missing",
+    "danger",
+    "error",
+    "bug",
+    "example",
+    "snippet",
+    "quote",
+    "cite",
+]
+
+
+def get_tag(params: str) -> Tuple[str, str]:
+    if not params.strip():
+        return "", ""
+
+    tag, *_title = params.strip().split(" ")
+    joined = " ".join(_title)
+
+    title = ""
+    if not joined:
+        title = tag.title()
+    elif joined != '""':
+        title = joined
+    return tag.lower(), title
+
+
+def validate(params: str) -> bool:
+    tag = params.strip().split(" ", 1)[-1] or ""
+    if tag.lower() in ADMONITION_TAGS:
+        return True
+    # FIXME: Should this return False? Or better way to log warnings?
+    print(f"Warning: admonition tag {tag} is not one of: {ADMONITION_TAGS}")
+    return bool(tag)
```

Comment:
TBD: How strict should the validation be? The original JS plugin didn't check against the list of tags